snuba/query/snql/__init__.py

```diff
@@ -0,0 +1,36 @@
+from snuba.datasets.dataset import Dataset
+from snuba.query.logical import Query
+from parsimonious.grammar import Grammar
+
+
+grammar = Grammar(
+    r"""
+    query_exp = "MATCH" overall_clause
+
+    overall_clause = match_clause where_clause? collect_clause? having_clause? order_by_clause?
+
+    match_clause = space* "(" clause ")" space*
+    where_clause = space* "WHERE" clause space*
+    collect_clause = space* "COLLECT" clause "BY" clause space*
+    having_clause = space* "HAVING" clause space*
```

Comment:
I assume these are going to change when you introduce the expressions from your previous grammar, is that right ?

Reply:
Yes, the plan is to modify each of the specific clauses and tailor them to their own structure (i.e. whether they include Expressions, etc).